Automattic/Automattic-Tracks-Android
# Allow clients to append data to an event before sending it

References #75 

This PR adds a feature to append data to every event before sending it.

It'll be primarily used for the `Encrypted Logging` feature but I've extended it to pass a `List<String>` so it doesn't limit to only one use case.

See https://github.com/wordpress-mobile/WordPress-Android/blob/a9eeed79f6d85fd3a3a55157ff9b2932248c00cc/WordPress/src/main/java/org/wordpress/android/util/CrashLogging.kt#L68-L79

### How to test?

1. Make sure there's a `sentryTestProjectDSN` in `gradle.properties` pointing to some playground Sentry project.
2. Run `SendEventsToSentry#shouldAppendExtraForKeyIfRequested`
3. Assert there's a new event with "key |  value" under `additional data` 
![image](https://user-images.githubusercontent.com/5845095/113865494-70594600-97ac-11eb-9de6-838dcb21f776.png)
